pkg/providers/vsphere/virtualmachine/publish.go

```diff
@@ -83,19 +84,23 @@ func CloneVM(
 		Value: cl.Spec.ID,
 	}
 
+	var moVM mo.VirtualMachine
 	vm := object.NewVirtualMachine(
 		vimClient,
 		vimtypes.ManagedObjectReference{
 			Type:  string(vimtypes.ManagedObjectTypeVirtualMachine),
 			Value: vmCtx.VM.Status.UniqueID,
 		})
 
-	var filteredExtraConfig pkgutil.OptionValues
-	var err error
-	if filteredExtraConfig, err = GetFilteredExtraConfigFromObject(
```

Comment:
Nit: can probably remove this function now since it's introduced for this call (not being used anywhere else).